### DISCUSSION TITLE
How to tell `vercel deploy --prebuilt` to only upload .next folder to avoid "Error: Invalid request: `files` should NOT have more than ..""

### DISCUSSION THREAD

**Discussion by giancarlosisasi** [vercel/vercel#8643] [Help] [answered]
Hi there
I'm using `vercel deploy --prebuilt -t $VERCEL_TOKEN --scope $VERCEL_SCOPE` to deploy my projects inside my own CI and after using `vercel pull` and `vercel build` we got the following error:

````
Error: Invalid request: `files` should NOT have more than 15000 items, received 17159.
````

As far as I know, the `--prebuilt` option should prevent uploading my source files to vercel. Instead, it should use the content of `.next` and `.vercel/output` (created by `vercel build`) to deploy my application.

Is there any option that I'm missing in the `vercel deploy --prebuilt` command?

**Comment by swarnava** [comment, marked as answer]
Hi,

Have you ran `vercel build` before using `vercel deploy --prebuilt` ? You can also pass `--archive=tgz` to reduce number of files to upload.

**Reply by giancarlosisasi** [reply]
hi @swarnava 
As I mentioned, I'm using `vercel pull`, `vercel build` and `vercel deploy --prebuilt`.

Also, is there any documentation/link where I can read more about the --archive=tgz?

**Reply by swarnava** [reply]
We don't have a concrete documentation for this yet. But you can a bit more: https://github.com/vercel/vercel/releases/tag/vercel%4028.0.0. basically it uploads the output file as single tarball instead of uploading files one by one

**Reply by giancarlosisasi** [reply]
@swarnava thanks but... the main problem here is that `vercel deploy --prebuilt` is not working correctly.

**Reply by swarnava** [reply]
Right. You are getting this error on vercel deploy, right? Not `vercel build`? And you are getting same error even with archive option?

**Reply by giancarlosisasi** [reply]
@swarnava yes, I think it's the same error in https://github.com/vercel/vercel/issues/8571

**Reply by swarnava** [reply]
I'll double check with team. Which version of vercel cli are you using?

**Reply by giancarlosisasi** [reply]
@swarnava `Vercel CLI 28.4.2`

**Reply by TooTallNate** [reply]
@GiancarlosIO What is the error that you are seeing exactly, specifically when doing `vercel deploy --archive=tgz --prebuilt`?

**Reply by giancarlosisasi** [reply]
@TooTallNate right now, I have tested the `--archive=tgz` (again) and it's working correctly, at least for preview deployments. Thanks for that!!!
At this point, just for security reasons, I just want to make sure that the `vercel deploy --prebuilt` is not uploading our source code to the vercel platform, is that right? or event with `prebuilt` it will upload our source code to vercel? 😢

**Reply by swarnava** [reply]
It doesn't upload source file. It stills upload output/build files in a single tar file. You can pass `--debug` to get more verbose log and see which file is being uploaded

**Reply by giancarlosisasi** [reply]
alright, thanks @swarnava !

**Comment by jzxhuang** [comment]
I'm facing a similar problem — tried the `--archive=tgz` trick, but running into a new issue:

some logs are truncated for privacy. I've also tried on latest CLI (28.15.0) and same issue occurs.

```
$ /root/myagent/_work/1/s/node_modules/.bin/vercel deploy --prebuilt --debug --token *** --archive tgz
Vercel CLI 28.4.11

[client-debug] 2023-02-02T17:33:59.233Z Creating deployment...
[client-debug] 2023-02-02T17:33:59.234Z Provided 'path' is a directory.
[client-debug] 2023-02-02T17:33:59.235Z Found 4 rules in .vercelignore
[client-debug] 2023-02-02T17:33:59.235Z Building file tree...
[client-debug] 2023-02-02T17:33:59.671Z Found 15633 files in the specified directory
[client-debug] 2023-02-02T17:33:59.671Z Packing tarball
[client-debug] 2023-02-02T17:35:58.266Z Packed tarball
> [debug] [2023-02-02T17:35:58.311Z] Error: RangeError: data is too long
RangeError: data is too long
    at Hash.update (node:internal/crypto/hash:113:22)
    at Object.hash (/root/myagent/_work/1/s/node_modules/vercel/dist/index.js:204115:40)
    at createDeployment (/root/myagent/_work/1/s/node_modules/vercel/dist/index.js:203625:30)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at processDeployment (/root/myagent/_work/1/s/node_modules/vercel/dist/index.js:239171:26)
    at Now.create (/root/myagent/_work/1/s/node_modules/vercel/dist/index.js:245967:28)
    at createDeploy (/root/myagent/_work/1/s/node_modules/vercel/dist/index.js:238777:16)
    at module.exports.__webpack_modules__.28640.exports.default (/root/myagent/_work/1/s/node_modules/vercel/dist/index.js:230133:22)
    at main (/root/myagent/_work/1/s/node_modules/vercel/dist/index.js:236140:20)
Error: data is too long
```

@swarnava @TooTallNate any ideas here? I'm having issues understanding how `vercel deploy` counts source files — is that the files in `.vercel/output`? Thanks!

**Reply by swarnava** [reply]
Hi, Looks like you already have a ticket with us `00120503`. We will continue conversation there. Thank you!
